ding/policy/impala.py

```diff
@@ -289,7 +314,15 @@ def _init_collect(self) -> None:
             Collect mode init method. Called by ``self.__init__``, initialize algorithm arguments and collect_model.
             Use multinomial_sample to choose action.
         """
-        self._collect_model = model_wrap(self._model, wrapper_name='multinomial_sample')
+
+        self._action_space = self._cfg.action_space
+        if self._action_space == 'continuous':
+            self._collect_model = model_wrap(self._model, wrapper_name='reparam_sample')
+        elif self._action_space == 'discrete':
+            self._collect_model = model_wrap(self._model, wrapper_name='multinomial_sample')
+        else:
```

Comment:
add assertion in the beginning of policy (init method), then you don't need to raise the error in following implementations.